src/qumin/representations/paradigms.py

```diff
@@ -69,7 +73,7 @@ def _get_unknown_segments(self, row, unknowns):
 
     def preprocess(self, fillna=True, segcheck=False,
                    defective=False, overabundant=False, merge_cols=False,
-                   cells=None, sample=None, most_freq=None, pos=None, **kwargs):
+                   cells=None, sample=None, sample_kws=dict(), pos=None, **kwargs):
```

Comment:
Nope, never create an object in the arguments: it'll be unique across all calls of the function !

Reply:
No, both are bad. You need to do sample_kws=None, then have a condition "if sample_kws is None: sample_kws = dict()" or something like that before you use it.

Reply:
Here's a description of this cursed thing:

https://www.pullrequest.com/blog/python-pitfalls-the-perils-of-using-lists-and-dicts-as-default-arguments/